woocommerce/woocommerce-android
# Issue/5997 Variations bulk update: regular price

Note: this PR implements the https://github.com/woocommerce/woocommerce-android/issues/5997 partially and targets the base feature branch.

### Description
👉🏻 This PR implements bulk update flow for variation's regular price.
* Added `VariationsBulkUpdatePriceFragment` and  `VariationsBulkUpdatePriceViewModel` classes responsible for handling price update operation. Those classes are implemented keeping in mind they will be used for sale price updates soon. 
* Updated `VariationRepository` to provide variations bulk update operation.
* Refactored logic around the `RegularPriceState` sealed class. Extracted to generic class: `ValuesGroupType` which will be reusable for attrs other than regular and sale prices too.

### Testing instructions
1. Open product details and go to "Variations" list (product must have variations defined).
2. After variations are loaded click the "Bulk update..." menu item.
3. The bottom sheet should appear. Choose "Regular price".
4. Enter the new regular price value and click "Done". 
5. Price update screen should be closed after successful operation and a snack should appear. In case of error the screen should not be closed and the error snack should be shown.

### Visual changes
| Light mode  | Dark mode |
| ------------- | ------------- |
|<video src="https://user-images.githubusercontent.com/4527432/169715484-3de45313-2dd4-441d-b352-38c8d541dda7.mp4"/>|<video src="https://user-images.githubusercontent.com/4527432/169715570-0745e616-e201-4557-bcba-bab620238b9a.mp4"/>|


### Roadmap
There are a few more things that should be implemented before the bulk update flow can be considered complete:
* Handling update in case the input field is empty. The value should be updated to "not set" like in other places in the app.
* Progress bar should be shown after clicking "done". The operation takes usually a few secs.
* Adding the "Undo" action to the confirmation snackbar. 
* Tracking "Done" button clicks.
* Handling bulk update for other properties: sale schedule, stock quantity, stock status, shipping class, weight, and dimensions.
* Unit-testing `VariationsBulkUpdatePriceViewModel`.
